src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -38,15 +51,52 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
         return configurations;
     }
 
+    public async afterResolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<void> {
+        const { task, promise } = netSdkRunTaskProvider.createNetSdkRunTask(
+            {
+                netCore: {
+                    appProject: await this.inferProjPath(context.actionContext, context.folder),
+                }
+            }
+        );
+        await tasks.executeTask(task);
+        await promise;
+    }
+
+    public async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
+        if (NetSdkDebugHelper.projPath) {
+            return NetSdkDebugHelper.projPath;
+        }
+
+        const projFileItem = await quickPickProjectFileItem(context, undefined, folder, 'No .csproj file could be found.');
+        NetSdkDebugHelper.projPath = projFileItem.absoluteFilePath; // save the path for future use
+        return projFileItem.absoluteFilePath;
+    }
+
     /**
-     * Checks if the launch task is using the .NET SDK Container build
-     * @param preLaunchTask
-     * @returns true if the launch task is using the .NET SDK Container build
-     *          false otherwise
+     * Overwite the base implementation to infer the container name from the project path (appProject)
+     * instead of the folder name
      */
-    public isDotNetSdkBuild(preLaunchTask: string): boolean {
-        return preLaunchTask === NetSdkTaskFullSymbol;
+    protected async loadExternalInfo(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<{ configureSsl: boolean, containerName: string, platformOS: PlatformOS }> {
+        NetSdkDebugHelper.projPath = debugConfiguration.netCore.appProject;
+
+        const associatedTask = context.runDefinition;
+        return {
+            configureSsl: !!(associatedTask?.netCore?.configureSsl),
+            containerName: this.inferDotNetSdkContainerName(debugConfiguration),
+            platformOS: associatedTask?.dockerRun?.os || 'Linux',
+        };
     }
+
+    public inferDotNetSdkContainerName(debugConfiguration: DockerDebugConfiguration): string {
+        const projFileUri = Uri.file(path.dirname(debugConfiguration.netCore.appProject));
+        return getDefaultContainerName(path.basename(projFileUri.fsPath), "dev");
+    }
+
+    public isDotNetSdkBuild(debugConfiguration: DockerDebugConfiguration): boolean {
```

Comment:
should we have this here or `NetCoreDebugHelper.ts`? my thought process was since it is related to .Net SDK debug, it should be in NetCoreDebugHelper. But at the same time, the only time it is being called is in `NetCoreDebugHelper.ts`. So.... idk?

Reply:
Added a couple comments on the existing usage of isDotNetSdkBuild; it would make sense to slightly refactor the code to use polymorphism to implement the SDK specific behavior rather than having SDK conditioned logic in the base NetCoreDebugHelper class. That would remove the need for isDotNetSdkBuild entirely.

Reply:
I love the idea!